### PR TITLE
add benchmark

### PR DESCRIPTION
**Pull request by stevemao**
Also add a loop implementation to be reference

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +0,0 @@
-
-test:
-	@node_modules/.bin/tape test/*.js
-
-.PHONY: test
-
```
Comment (juliangruber) [inline Makefile]: i like my makefiles, can you leave this in place?

**Comment by juliangruber** [comment]
otherwise +1, and a great idea to add an alternative implementation for reference!

**Comment by stevemao** [comment]
Do you want to use makefile to start benchmark too?

**Comment by juliangruber** [comment]
yes that would be preferred

**Comment by stevemao** [comment]
@juliangruber can you have a look again?

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 test:
-	@node_modules/.bin/tape test/*.js
+	@node_modules/.bin/tape test/test.js
 
 .PHONY: test
```
Comment (juliangruber) [inline Makefile]: the `bench` target needs to be phony too, so this line should be `.PHONY: test bench`. in a makefile you mark a target as phony when it doesn't depend on file changes to be executed, like a c recompile, but rather is executed when a developer sees fit

**Comment by stevemao** [comment]
Hope I got it right this time

**Comment by juliangruber** [comment]
yep, looks great!

**Pull request merged**